ui/src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useRef } from 'react'
+import styles from './Modal.module.scss'
+import PropTypes from 'prop-types'
+import { ButtonIcon } from '../ButtonIcon/ButtonIcon'
+import Button from '../Button/Button'
+
+/**
+ * ### example:
+ * ```
+ * import Modal from 'components/Modal/Modal'
+ * const [open, setOpen] = useState(false)
+ * return (
+ *   <>
+ *     <button
+ *       onClick={() => {
+ *         setOpen(true)
+ *       }}
+ *     >
+ *       Open Modal
+ *     </button>
+ *     <Modal open={open} onClose={() => setOpen(false)} />
+ *   </>
+ * )
+ * ```
+ * ### props:
+ * - title - title of the modal
+ * - children - content of the modal
+ * - variant - type of modal
+ *  - default - standard modal
+ *  - confirm - modal with confirm button
+ * - modal - boolean value that controls modal type
+ *  - true - modal, default value
+ *  - false - dialog, allows interaction with the rest of the page
+ * - open - state of the modal
+ * - onClose - function to change state of the modal
+ *
+ * ### reference:
+ * - [dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog)
+ * */
+function Modal({ title, children, modal, variant, open, onClose }) {
+  let showConfirm
```

Comment:
Możesz tutaj użyć `ternary operator` i usunąć tego `let`a. Chyba, że masz jakiś głębszy zamysł z użyciem `let` w tym przypadku 😅
```suggestion
  const showConfirm = variant === "confirm" ? true : false;
```

Reply:
Zrobiłem tak w razie wypadku gdyby w przyszłości miałoby się pojawić więcej wariantów, niżej jest `switch`, który by tym potem sterował